service/hook/assembla/assembla_test.go

```diff
@@ -0,0 +1,202 @@
+package assembla
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/bitrise-io/bitrise-webhooks/bitriseapi"
+	"io/ioutil"
+	"strings"
+)
+
+func Test_detectContentType(t *testing.T) {
+	t.Log("Push event - should handle")
+	{
+		header := http.Header{
+			"Content-Type":     {"application/json"},
+		}
+		contentType, err := detectContentType(header)
+		require.NoError(t, err)
+		require.Equal(t, "application/json", contentType)
+	}
+}
+
+func Test_transformPushEvent(t *testing.T) {
+	t.Log("Do Transform - code push")
+	{
+		pushEvent := PushEventModel{
+			SpaceEventModel: SpaceEventModel{
+				Space: "Space name",
+				Action: "committed",
+				Object: "Changeset",
+			},
+			MessageEventModel: MessageEventModel{
+				Title: "1 commits [branchname]",
+				Body: "ErikPoort pushed 1 commits [branchname]\n",
+				Author: "ErikPoort",
+			},
+			GitEventModel: GitEventModel{
+				RepositorySuffix: "origin",
+				RepositoryURL: "git@git.assembla.com:username/project.git",
+				Branch: "branchname",
+				CommitID: "sha1chars11",
+			},
+		}
+
+		// OK
+		{
+			hookTransformResult := transformPushEvent(pushEvent)
+			err := detectAssemblaData(pushEvent)
+			require.Equal(t, err, nil)
+			require.NoError(t, hookTransformResult.Error)
+			require.False(t, hookTransformResult.ShouldSkip)
+			require.Equal(t, []bitriseapi.TriggerAPIParamsModel{
+				{
+					BuildParams: bitriseapi.BuildParamsModel{
+						CommitMessage: "ErikPoort pushed 1 commits [branchname]\n",
+						Branch:        "branchname",
+					},
+					TriggeredBy: "ErikPoort",
```

Comment:
`TriggeredBy` should be `"webhook"` right now. We'll add "author" infos a bit later, right now the `TriggeredBy` indicates the source *type* (webhook, manual, curl, ...) not the exact author.

If you're interested feel free to create a feature request for tracking the status of the "author" info at https://discuss.bitrise.io/c/feature-request so that we can keep you up to date with the progress ;)